src/interpreter/kernel/instruction/binop.js

```diff
@@ -9,52 +10,116 @@ const f64 = require('../../runtime/values/f64');
 
 // https://webassembly.github.io/spec/exec/instructions.html#exec-binop
 function binop(
-  c1: StackLocal,
-  c2: StackLocal,
+  {value: c1} : StackLocal,
+  {value: c2}: StackLocal,
   sign: Sign,
   createValue: (number) => StackLocal
 ): StackLocal {
 
   switch (sign) {
   // https://webassembly.github.io/spec/exec/numerics.html#op-iadd
   case 'add':
-    return createValue(c1.value + c2.value);
+    return createValue(c1 + c2);
 
-    // https://webassembly.github.io/spec/exec/numerics.html#op-isub
+  // https://webassembly.github.io/spec/exec/numerics.html#op-isub
   case 'sub':
-    return createValue(c1.value - c2.value);
+    return createValue(c1 - c2);
 
   // https://webassembly.github.io/spec/exec/numerics.html#op-imul
   case 'mul':
-    return createValue(c1.value * c2.value);
+    return createValue(c1 * c2);
 
   // https://webassembly.github.io/spec/exec/numerics.html#op-idiv-u
   // https://webassembly.github.io/spec/exec/numerics.html#op-idiv-s
   case 'div_s':
   case 'div_u':
   case 'div':
-    return createValue(c1.value / c2.value);
+    return createValue(c1 / c2);
 
   // https://webassembly.github.io/spec/exec/numerics.html#op-iand
   case 'and':
-    return createValue(c1.value & c2.value);
+    return createValue(c1 & c2);
 
   // https://webassembly.github.io/spec/exec/numerics.html#op-ior
   case 'or':
-    return createValue(c1.value | c2.value);
+    return createValue(c1 | c2);
 
   // https://webassembly.github.io/spec/exec/numerics.html#op-ixor
   case 'xor':
-    return createValue(c1.value ^ c2.value);
+    return createValue(c1 ^ c2);
 
+  // https://webassembly.github.io/spec/exec/numerics.html#op-fmin
   case 'min':
-    return createValue(Math.min(c1.value, c2.value));
+    return createValue(Math.min(c1, c2));
 
+  // https://webassembly.github.io/spec/exec/numerics.html#op-fmax
   case 'max':
-    return createValue(Math.max(c1.value, c2.value));
+    return createValue(Math.max(c1, c2));
 
+  // https://webassembly.github.io/spec/exec/numerics.html#op-fcopysign
   case 'copysign':
-    return createValue((Math.sign(c1.value) === Math.sign(c2.value)) ? c1.value : -c1.value);
+    return createValue((Math.sign(c1) === Math.sign(c2)) ? c1 : -c1);
+  }
+
+  throw new Error('Unsupported binop: ' + sign);
+}
+
+function binopLong(
+  {value: c1} : StackLocal,
+  {value: c2} : StackLocal,
+  sign: Sign,
+  createValue: (LongNumber) => StackLocal
+): StackLocal {
+  
+  let longResult: LongNumber;
+
+  switch (sign) {
+  // https://webassembly.github.io/spec/exec/numerics.html#op-iadd
```

Comment:
Specs links are broken here, but I can update them all at once (search and replace probably). https://github.com/xtuc/js-webassembly-interpreter/issues/80